server/src/main/java/org/opensearch/identity/IdentityService.java

```diff
@@ -56,4 +63,41 @@ public Subject getSubject() {
     public TokenManager getTokenManager() {
         return identityPlugin.getTokenManager();
     }
+
+    public static IdentityService getInstance() {
+        if (instance == null) {
+            new IdentityService(Settings.EMPTY, List.of());
+        }
+        return instance;
+    }
+
+    // TODO: Find a way to combine these
+
+    public Set<String> getActionScopes(Principal principal) {
```

Comment:
This and the method don't appear to be used, part of scope validation that can be done seperately?

Reply:
Yeah gonna get rid of these.